backend/src/user/structures/Uuid.ts

```diff
@@ -0,0 +1,16 @@
+import { Record, Static, String } from "runtypes";
```

Comment:
Nie do końca mi się to widzi, żeby kluczem obcym w tych projektach była struktura `{id, provider}`, to może utrudniać tworzenie jakiś zapytań w przyszłości. Ja bym proponował może coś takiego, że kolekcja user w bazie danych będzie miała takie pola `{_id, credentials:{id, provider}, username, email, picture}`. To _id będzie unikalnym kluczem nadawanym przez mongo i to do niego będziemy mogli robić powiązanie w projektach. Pole credentials (roboczo nazwane można dać inną nazwę) będzie używane tylko przy autentykacji, natomiast dalej w aplikacji już tego nie używamy.

Reply:
Nie wiem czy jak mielibyśmy robić osobne _id to czy jest sens trzymać id z usługi zewnętrznej. Cały koncept z tym 
`{id, provider}` był taki żeby mieć unique constraint bez wprowadzania nowych pól, ale id usługi zewnętrznej chyba nie jest do niczego używane samo z siebie.